### PR TITLE
step 10

### PR DESCRIPTION
**Pull request by andriimikhaylovskiy**

### CODE REVIEW

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Your build was failed

File: pom.xml [hunk 1]
```diff
@@ -15,81 +15,106 @@
...
 			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>org.mapstruct</groupId>
-			<artifactId>mapstruct</artifactId>
-			<version>1.5.3.Final</version>
+			<groupId>com.puppycrawl.tools</groupId>
+			<artifactId>checkstyle</artifactId>
+			<version>10.15.0</version>
```
Comment (sarakhmen) [inline pom.xml]: do you really need this dependency?

File: src/main/java/mate/academy/spring/boot/repository/BookRepository.java [hunk 2]
```diff
@@ -2,9 +2,16 @@
...
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
     List<Book> findAllByAuthorContainsIgnoreCase(String author);
+
+    @Query(value = "SELECT * FROM books b INNER JOIN books_categories bc ON "
+            + "b.id = bc.book_id WHERE bc.category_id = :categoryId", nativeQuery = true)
+    List<Book> findAllByCategorySet_Id(@Param("categoryId") Long categoryId, Pageable pageable);
```
Comment (sarakhmen) [inline src/main/java/mate/academy/spring/boot/repository/BookRepository.java]: please follow Java's method naming convention

File: src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java [hunk 3]
```diff
@@ -23,13 +24,15 @@ public class BookServiceImpl implements BookService {
     private final BookMapper bookMapper;
     private final Random myRandom;
     private final BookSpecificationBuilder bookSpecificationBuilder;
+    //private final CategoryRepository categoryRepo;
```
Comment (sarakhmen) [inline src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java]: remove commented field

**Review by sarakhmen — CHANGES_REQUESTED** [review]
good job but let's slightly improve your solution :)

**Review by Elena-Bruyako — APPROVED** [review]
Good job!

**Pull request merged**